form/src/Components/SignIn.js

```diff
@@ -1,17 +1,86 @@
-import React from "react";
+import React, { useState } from "react";
 
 const SignIn = ({ firebase, auth }) => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [formDisplay, setFormDisplay] = useState("none")
+
   const signInWithGoogle = () => {
     const provider = new firebase.auth.GoogleAuthProvider();
     auth.signInWithPopup(provider);
   };
 
+  const signInWithEmail = (event) => {
+    event.preventDefault();
+    firebase.auth().signInWithEmailAndPassword(email, password)
+  }
+
+  const signUpWithEmail = (event) => {
+    event.preventDefault();
+    firebase.auth().createUserWithEmailAndPassword(email, password)
+  }
+
+  const handleDisplay = (display) => {
+    setPassword("");
+    setEmail("");
+    setFormDisplay(display)
+  }
+
   return (
-    <>
+    <div class="center sign-in">
       <button className="btn btn-signin" onClick={signInWithGoogle}>
-        Sign in with Google
+        Sign in or sign up with Google
       </button>
-    </>
+      {
```

Comment:
i think this looks too convoluted and messy. lmk if you have any ideas on how to improve it! i'll do some research later too. @siam132